person.rb

```diff
@@ -0,0 +1,29 @@
+require_relative './nameable'
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..10_000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    super()
+  end
+
+  def can_use_services?
+    @parent_permission || is_of_age?
+  end
+
+  def correct_name
+    @name
+  end
+
+  ############ private methods ############
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
```

Comment:
- [ ] You are missing the method for implementing the other end of the `many-to-many` relationship for the **`Person`**, Kindly add it here as you have in the **`Book`** class `(add_rental)`.

  Remember to ensure uniformity of parameters.